### PR TITLE
Adds ConnectionInfo and SecureSession to Ember

### PR DESCRIPTION
**Pull request by rossabaker**
For parity with the Blaze and Servlet backends.

### CODE REVIEW

File: ember-server/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala [hunk 1]
```diff
@@ -175,40 +182,82 @@ private[server] object ServerHelpers {
...
+          Request.Connection(
+            local = local,
+            remote = remote,
+            secure = socket.isInstanceOf[TLSSocket[F]]
+          ))
+      case _ =>
+        Vault.empty
+    }
+
+  private def mkSecureSession[F[_]: Applicative](socket: Socket[F]) =
```
Comment (rossabaker) [inline ember-server/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala:247]: Confession: I haven't tested this one except with types.

File: ember-server/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala [hunk 2]
```diff
@@ -175,40 +182,82 @@ private[server] object ServerHelpers {
...
+    Stream.eval(mkRequestVault(socket)).flatMap { requestVault =>
+      Stream
+        .unfoldLoopEval(socket.reads(receiveBufferSize, durationToFinite(idleTimeout)))(s =>
+          runApp(
+            s,
+            maxHeaderSize,
+            requestHeaderReceiveTimeout,
+            httpApp,
+            errorHandler,
+            requestVault).attempt.map {
```
Comment (rossabaker) [inline ember-server/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala:195]: Adding `requestVault` is the only change in this hunk.  The rest is indentation from the flatMapping of `requestVault`

File: core/src/main/scala/org/http4s/internal/tls.scala [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    else if (cipherSuite.contains("WITH_AES_256_")) 256
+    else if (cipherSuite.contains("WITH_RC4_128_")) 128
+    else if (cipherSuite.contains("WITH_AES_128_")) 128
+    else if (cipherSuite.contains("WITH_RC4_40_")) 40
+    else if (cipherSuite.contains("WITH_3DES_EDE_CBC_")) 168
+    else if (cipherSuite.contains("WITH_IDEA_CBC_")) 128
+    else if (cipherSuite.contains("WITH_RC2_CBC_40_")) 40
+    else if (cipherSuite.contains("WITH_DES40_CBC_")) 40
+    else if (cipherSuite.contains("WITH_DES_CBC_")) 56
+    else 0
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/internal/tls.scala:77]: This 0 seems weird but it's been like that.

**Comment by tpolecat** [comment]
Thanks for this!

**Review by RaasAhsan — APPROVED** [review]

**Pull request merged**